utils/CI/check_wml_images.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+#
+# Find missing images referenced by WML (as in issue #8432)
+#
+# Requires sort with -s option (stable sort, disable further byte-by-byte
+# comparison of lines that collate equally by specified sort key), e.g. from
+# GNU coreutils
+#
+# Example output:
+#     data/multiplayer/scenarios/2p_Aethermaw.cfg:134: missing image: units/undead/shadow-s-attack-1.png
+#     data/multiplayer/scenarios/2p_Aethermaw.cfg:135: missing image: units/undead/shadow-n-attack-2.png
+#     data/multiplayer/scenarios/2p_Aethermaw.cfg:147: missing image: units/undead/shadow-n-3.png
+#     data/multiplayer/scenarios/2p_Aethermaw.cfg:149: missing image: units/undead/shadow-s-attack-4.png
+#     data/multiplayer/scenarios/2p_Aethermaw.cfg:196: missing image: units/human-loyalists/spearman-attack-se-10.png
+#     data/multiplayer/scenarios/2p_Aethermaw.cfg:197: missing image: units/human-loyalists/spearman-attack-s-6.png
+#     data/multiplayer/scenarios/2p_Aethermaw.cfg:198: missing image: units/human-loyalists/spearman-attack-se-9.png
+#     data/multiplayer/scenarios/2p_Aethermaw.cfg:203: missing image: units/human-loyalists/general-idle-5.png
+
+set -eu
+
+# Parsing WML with a regular expression isn't ideal, but it's easy and it works.
+IMG_RE='^[^#]*[{]PLACE_IMAGE[^ ]* +[(]?"?([^{}$ ~")]+)[ ~")].*$'
+
+# In `! { COMMANDS | grep -- '.'; }` below, `grep -- '.'` returns a successful
+# exit status if `COMMANDS` produce output, and `!` inverts it into an error
+# status.  A less clever/obscure solution would be `es=true` before the loop,
+# `es=false` inside the loop after all the `continue`s, and `${es}` at the end;
+# but that won't work because all commands in pipelines run in subshells that
+# can't affect variables in the parent shell.
```

Comment:
I feel Python is probably the answer here, although I'd understand totally if you said that that would involve someone else rewriting the utility.

Reply:
I'm an expert in (POSIX) shell command language, and Wesnoth has 39 other shell scripts (including all 7 existing files in utils/CI/), so it seemed like the natural choice.  I tried to document the script enough to be maintainable (hence that comment), but feel free to suggest improvement in other parts of the (very short) code that may be unclear.

On the other hand I know very little Python, so if I had to rewrite it, that could take me a little more time.

Reply:
My `!` and `grep` trick is simple enough that another language like Python didn't occur to me as necessary, but it felt obscure enough that I should document its purpose for future readers.  In general, writing more comments and commit message text than code as I did here is not at all unusual for me. :)

Also worth noting that this script only finds `data/core/macros/image-utils.cfg`'s `PLACE_IMAGE` and `PLACE_IMAGE_SUBMERGED` macros, not other occurrences of images like those updated by #8437 (those however weren't missing).  Perhaps it should also look for lines like (pseudo-pattern/RE) `*="filename.(jpg|png|webp)"`.